sql/group.go

```diff
@@ -99,6 +103,7 @@ type groupNode struct {
 	render    []parser.Expr
 	funcs     []*aggregateFunc
 	needGroup bool
+	seen      []map[string]struct{}
```

Comment:
Rather than a separate `seen` slice that parallels the `funcs` slice, could you place this map inside `aggregateFunc`?